pupil_src/shared_modules/video_capture/file_backend.py

```diff
@@ -187,31 +181,36 @@ def _get_containers(self, index):
         try:
             container = av.open(self.containers_path[index])
         except av.AVError:
-            logger.info("The vide is broken")
-            self.current_index += 1
-            return self._get_containers(self.current_index)
+            logger.info(
+                "Video at {0} is broken".format(self.containers_path[index]))
+            # return the next container
+            self.current_container_index += 1
+            return self._get_containers(self.current_container_index)
         except IndexError:
-            logger.info("No More Container Found")
+            logger.info("No more container found")
             raise EndofVideoError("Reached end of video file")
         else:
             return container
 
-    def _get_timestamps_lst(self, source_path):
-        # import pdb; pdb.set_trace()
-        timestamps = self._get_pattern_lst(source_path, timestamps=True)
+    def _set_timestamps(self, source_path):
+        self.timestamps_lst = self._get_pattern_lst(
+            source_path, timestamps=True)
         avg_rate = self.video_stream.average_rate
         if avg_rate is None:
             avg_rate = Fraction(0, 1)
 
         if float(avg_rate) % 1 != 0.0:
             logger.error(
-                "Videofile pts are not evenly spaced, pts to index conversion may fail and be inconsitent."
+                "Videofile pts are not evenly spaced, pts to" +
+                "index conversion may fail and be inconsitent."
             )
         try:
-            self.timestamps = np.load(timestamps[0])
+            self.timestamps = np.load(self.timestamps_lst[0])
```

Comment:
I have just realised that we will require the timestamps of all partial timestamp files in `self.timestamps`.

Reply:
Yes, for displays the whole time of the bar at the bottom of the screen, right? I will use other var to store the timestamp of the current video later.

Reply:
Yes, and some other places, too.

Reply:
Ok, I see. so self.timestamps should contain all timestamp from the different part of videos. And we use another var to store the timestamp of the current video.

Reply:
Yes, I think this is a good solution.